clientv3/integration/watch_test.go

```diff
@@ -396,3 +396,80 @@ func TestWatchCompactRevision(t *testing.T) {
 		t.Fatalf("expected closed channel, but got %v", wresp)
 	}
 }
+
+func TestWatchWithProgressNotify(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	tests := []struct {
+		currentRevision int64
+		events          []*storagepb.Event
+		compactRevision int64
+		canceled        bool
+	}{
+		{1, []*storagepb.Event{}, 0, false},
+		{1, []*storagepb.Event{}, 0, false},
+		{1, []*storagepb.Event{}, 0, false},
+
+		{2, []*storagepb.Event{{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: []byte("key_0"), Value: []byte("value"), CreateRevision: 2, ModRevision: 2, Version: 1},
+		}}, 0, false},
+
+		{3, []*storagepb.Event{{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: []byte("key_1"), Value: []byte("value"), CreateRevision: 3, ModRevision: 3, Version: 1},
+		}}, 0, false},
+
+		{3, []*storagepb.Event{}, 0, false},
+		{3, []*storagepb.Event{}, 0, false},
```

Comment:
That's fine, but I still think it's racey--
Client
1. receive 3rd notification rev=1
2. close(ready)
3. start Puts

Server:
1. send 3rd notification rev=1
2. notification timeout; send 4th notification at rev=1
3. receive PUT rev=2